src/OrbitBase/include/OrbitBase/Typedef.h

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_TYPEDEF_H_
+#define ORBIT_BASE_TYPEDEF_H_
+
+#include <functional>
+#include <type_traits>
+#include <utility>
+
+namespace orbit_base {
+
+template <typename T, typename U>
+using EnableIfUConvertibleToT = std::enable_if_t<std::is_convertible_v<U, T>>;
+
+template <typename T, typename U>
+using EnableIfUIsNotConvertibleToT = std::enable_if_t<!std::is_convertible_v<U, T>>;
+
+// Strong typedef.
+// It is parameterized by two types. `T`, which represents the type of the stored value and `Tag_`,
+// which allows to distinguish between different Typedefs. First, the user is expected to define a
+// tag. Like `struct TestTag {};`
+// Also, a templated alias can come in handy
+// ```
+// template <typename T>
+// using Wrapper = Typedef<TestTag, T>;
+// ```
+// Now we can instantiate the typedefs
+// ```
+// const Wrapper<int> kFirstWrapped(1);
+// const Wrapper<int> kSecondWrapped(2);
+// ```
+// and define a callable (not necessarily a lambda)
```

Comment:
I think there's a bit of a conceptual jump here. It sounds like the only point of `Typedef` is `Apply`. It should be one paragraph describing `Typedef` and one describing `Apply`.

Reply:
I'll just move to the comment above `Apply`